mtags/src/main/scala-2/scala/tools/nsc/interactive/MetalsGlobalThread.scala

```diff
@@ -1,67 +0,0 @@
-package scala.tools.nsc.interactive
-
-import scala.meta.internal.pc.InterruptException
-
-/**
- * Adapation of PresentationCompilerThread from scala/scala.
- *
- * Changes are marked with "+- scalac deviation" comments.
- * Note that most of the work with the presentation compiler does not happen on
- * this thread, see CompilerJobQueue.
- */
-final class MetalsGlobalThread(var compiler: Global, name: String = "")
-    extends Thread("Scala Presentation Compiler [" + name + "]") {
-
-  /**
-   * The presentation compiler loop.
-   */
-  override def run(): Unit = {
-    compiler.debugLog("starting new runner thread")
-    while (compiler ne null) try {
-      compiler.checkNoResponsesOutstanding()
-      compiler.log.logreplay(
-        "wait for more work", {
-          compiler.scheduler.waitForMoreWork(); true
-        }
-      )
-      compiler.pollForWork(compiler.NoPosition)
-      while (compiler.isOutOfDate) {
-        try {
-          compiler.backgroundCompile()
-        } catch {
-          case _: FreshRunReq =>
-            compiler.debugLog("fresh run req caught, starting new pass")
-        }
-        compiler.log.flush()
-      }
-    } catch {
-      case ShutdownReq =>
-        compiler.debugLog("exiting presentation compiler")
-        compiler.log.close()
-
-        // make sure we don't keep around stale instances
-        compiler = null
-      case ex: Throwable =>
-        compiler.log.flush()
-
-        ex match {
-          // + scalac deviation
-          case InterruptException() =>
```

Comment:
We no longer do interrupt the presentation compiler thread itself, so this should no longer be needed. Handling `ThreadDeath` would probably be still useful, but it doesn't really help us to have it here as the presentation compiler thread is still running anyway.